domain/removal/state/model/relationwithremoteconsumer.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package model
+
+import (
+	"context"
+	"time"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/collections/transform"
+
+	"github.com/juju/juju/domain/removal"
+	"github.com/juju/juju/domain/removal/internal"
+	"github.com/juju/juju/internal/errors"
+)
+
+// RelationWithRemoteConsumerExists returns true if a relation exists with the input
+// UUID, and relates a synthetic application
+func (st *State) RelationWithRemoteConsumerExists(ctx context.Context, rUUID string) (bool, error) {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return false, errors.Capture(err)
+	}
+
+	remoteRelationUUID := entityUUID{UUID: rUUID}
+	existsStmt, err := st.Prepare(`
+SELECT r.uuid AS &entityUUID.uuid
+FROM   relation AS r
+JOIN   relation_endpoint AS re ON r.uuid = re.relation_uuid
+JOIN   application_endpoint AS ae ON re.endpoint_uuid = ae.uuid
+JOIN   application AS a ON ae.application_uuid = a.uuid
+JOIN   charm AS c ON a.charm_uuid = c.uuid
+JOIN   charm_source AS cs ON c.source_id = cs.id
+WHERE  r.uuid = $entityUUID.uuid
+AND    cs.name = 'cmr'`, remoteRelationUUID)
+	if err != nil {
+		return false, errors.Errorf("preparing remote relation exists query: %w", err)
+	}
+
+	var remoteRelationExists bool
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err = tx.Query(ctx, existsStmt, remoteRelationUUID).Get(&remoteRelationUUID)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return nil
+		} else if err != nil {
+			return errors.Errorf("running remote relation exists query: %w", err)
+		}
+
+		remoteRelationExists = true
+		return nil
+	})
+
+	return remoteRelationExists, errors.Capture(err)
+}
+
+// EnsureRelationWithRemoteConsumerNotAliveCascade ensures that the relation identified
+// by the input UUID is not alive, and sets the synthetic units in scope
+// of this relation to dead.
```

Comment:
Add a note about why this is ok i.e. because synthetic units don't have a uniter